Source/WebCore/style/values/primitives/StylePrimitiveNumericTypes.h

```diff
@@ -197,390 +195,140 @@ template<CSS::Range R = CSS::All> struct Flex {
     constexpr bool operator==(ValueType other) const { return value == other; };
 };
 
-// MARK: Dimension + Percentage Primitives
-
-template<typename T> struct PrimitiveDimensionPercentageCategory;
-
-template<auto R> struct PrimitiveDimensionPercentageCategory<Angle<R>> {
-    static constexpr auto category = Calculation::Category::AnglePercentage;
-};
+} // namespace Style
+} // namespace WebCore
 
-template<auto R> struct PrimitiveDimensionPercentageCategory<Length<R>> {
-    static constexpr auto category = Calculation::Category::LengthPercentage;
+namespace WTF {
+ 
+// Allow primitives numeric types that usually stored their value as a `double` to be
+// used with CompactVariant by using a `float`, rather than `double` representation
```

Comment:
But why is it OK to lose range and precision? What makes this lossless?

Reply:
Nothing makes it lossless, but it matches the existing behavior we have for LengthPercentage types in the WebCore::Length class. We may want to reconsider it at some point, but for now, it is at least consistent.